master/buildbot/process/properties.py

```diff
@@ -421,6 +421,27 @@ def getRenderingFor(self, build):
 
 
 @implementer(IRenderable)
+class _SecretRenderer(object):
+
+    def __init__(self, secret_name):
+        self.secret_name = secret_name
+
+    def getRenderingFor(self, build):
+        credsservice = build.getBuild().master.namedServices['secrets']
```

Comment:
what if secret manager is not configured? I think you should add a  .get() and if credsservice is not None